ludwig/schema/hyperopt/scheduler.py

```diff
@@ -0,0 +1,641 @@
+    )
+
+    stop_last_trials: bool = schema_utils.Boolean(
+        default=True, description=("Whether to terminate the trials after reaching max_t. Defaults to True.")
+    )
+
+
+# TODO: Double-check support for this
+@register_scheduler_config("fifo")
+@dataclass
+class FIFOSchedulerConfig(BaseSchedulerConfig):
+    """FIFO trial scheduler settings."""
+
+    type: str = schema_utils.StringOptions(options=["fifo"], default="fifo", allow_none=False)
+
+
+# TODO: Double-check support for this as well as whether Callable args work properly
+@register_scheduler_config("resource_changing")
+@dataclass
+class ResourceChangingSchedulerConfig(BaseSchedulerConfig):
+    """Resource changing scheduler settings."""
+
+    type: str = schema_utils.StringOptions(options=["resource_changing"], default="resource_changing", allow_none=False)
+
+    base_scheduler: Union[str, None, Callable] = schema_utils.String(
+        description=("The scheduler to provide decisions about trials. If None, a default FIFOScheduler will be used.")
+    )
+
+    resources_allocation_function: Union[str, Callable] = schema_utils.String(
+        description=(
+            "The callable used to change live trial resource requiements during tuning. This callable will be called on"
+            " each trial as it finishes one step of training. The callable must take four arguments: TrialRunner, "
+            "current Trial, current result dict and the ResourceChangingScheduler calling it. The callable must return "
+            "a PlacementGroupFactory, Resources, dict or None (signifying no need for an update). If "
+            "resources_allocation_function is None, no resource requirements will be changed at any time. By default, "
+            "DistributeResources will be used, distributing available CPUs and GPUs over all running trials in a robust"
+            " way, without any prioritization."
+        )
+    )
+
+
+def get_scheduler_conds():
+    """Returns a JSON schema of conditionals to validate against optimizer types defined in
+    `ludwig.modules.optimization_modules.optimizer_registry`."""
+    conds = []
+    for scheduler_config in scheduler_config_registry:
+        scheduler_cls = scheduler_config_registry[scheduler_config]
+        other_props = schema_utils.unload_jsonschema_from_marshmallow_class(scheduler_cls)["properties"]
+        schema_utils.remove_duplicate_fields(other_props)
+        preproc_cond = schema_utils.create_cond(
+            {"type": scheduler_config},
+            other_props,
+        )
+        conds.append(preproc_cond)
+    return conds
+
+
+def SchedulerDataclassField(default={"type": "async_hyperband"}, description="Hyperopt scheduler settings."):
+    """Custom dataclass field that when used inside of a dataclass will allow any scheduler in
+    `ludwig.schema.hyperopt.scheduler.scheduler_registry`.
+
+    Sets default optimizer to 'async_hyperband'.
```

Comment:
Looks like there are a handful of copy/paste remainders from the optimizer here, may want to `ctrl+f` and address all of them so the docstring makes sense.

Reply:
+1